readthedocs/proxito/views/hosting.py

```diff
@@ -0,0 +1,128 @@
+"""Views for hosting features."""
+
+import structlog
+from django.conf import settings
+from django.http import JsonResponse
+from django.views import View
+
+from readthedocs.builds.constants import EXTERNAL
+from readthedocs.core.mixins import CDNCacheControlMixin
+from readthedocs.core.resolver import resolver
+from readthedocs.core.unresolver import unresolver
+
+log = structlog.get_logger(__name__)  # noqa
+
+
+class ReadTheDocsConfigJson(CDNCacheControlMixin, View):
+    def get(self, request):
+
+        url = request.GET.get("url")
+        if not url:
+            return JsonResponse(
+                {"error": "'url' GET attribute is required"},
+                status=400,
+            )
+
+        unresolved_domain = request.unresolved_domain
+        project = unresolved_domain.project
+
+        # TODO: why the UnresolvedURL object is not injected in the `request` by the middleware.
+        # Is is fine to calculate it here?
```

Comment:
Only the unresolved domain is injected for all proxito views, the unresolved URL is needed only when serving docs, so you need to call it manually only when you need it. And you are parsing the URL from a query param, so you need to call it manually anyway.